### PR TITLE
Random Forest, Gradient Boosting: Handle deprecated parameters

### LINKED ISSUE #6217: Deprecate:  `max_features='auto'
```test_scorer_feature (Orange.tests.test_random_forest.RandomForestTest) ... /home/runner/work/orange3/orange3/.tox/orange-released/lib/python3.8/site-packages/sklearn/ensemble/_forest.py:427: FutureWarning: `max_features='auto'` has been deprecated in 1.1 and will be removed in 1.3. To keep the past behaviour, explicitly set `max_features='sqrt'` or remove this parameter as it is also the default value for RandomForestClassifiers and ExtraTreesClassifiers.```

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

Fixes #6217, #6218, #6219

##### Description of changes
- set default RandomForestLearner.max_features to `sqrt`
- set default RandomForestRegressionLearner.max_features to `1.0`
- set default GBClassifier.loss to `log_loss`
- remove `use_label_encoder` parameter from XGBClassifier 
- remove `use_label_encoder` parameter from XGBRFClassifier
- set catboost minimum version

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Review by markotoplak — CHANGES_REQUESTED** [review]
Random forest defaults do not keep previous behavior.

For classification, it should be "sqrt" (and no old tests should change outputs).

For regression, `1.0` is correct but note that it means "use all features". Therefore it is likely misleading to change the spinbox limits unless scikit-learn handles `1` and `1.0` differently.

**Review by markotoplak — APPROVED** [review]

**Comment by markotoplak** [comment]
Thanks, @VesnaT. I see that we do not have oldest tests or required versions defined (requirements-opt) for xgboost and catboost. Now it is a good time to add them.
